src/Caching/Impl/Models/ModuleModel.cs

```diff
@@ -69,6 +70,9 @@ internal sealed class ModuleModel : MemberModel {
             var classes = new Dictionary<string, ClassModel>();
             var typeVars = new Dictionary<string, TypeVarModel>();
             var namedTuples = new Dictionary<string, NamedTupleModel>();
+            var superTypes = new Dictionary<string, SuperModel>();
+
+            superTypes["super"] = new SuperModel();
```

Comment:
Since the entire super type can be described by its qualified name, I don't think you actually need to end up with a model for it, just create the type on the fly during the restore.